homeassistant/components/life360/device_tracker.py

```diff
@@ -1,432 +1,267 @@
+from homeassistant.components.device_tracker import SOURCE_TYPE_GPS
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
     ATTR_BATTERY_CHARGING,
-    ATTR_ENTITY_ID,
+    ATTR_BATTERY_LEVEL,
+    ATTR_ENTITY_PICTURE,
+    ATTR_GPS_ACCURACY,
+    ATTR_LATITUDE,
+    ATTR_LONGITUDE,
+    ATTR_NAME,
     CONF_PREFIX,
-    LENGTH_FEET,
-    LENGTH_KILOMETERS,
-    LENGTH_METERS,
-    LENGTH_MILES,
-    STATE_UNKNOWN,
 )
-from homeassistant.core import HomeAssistant
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.event import track_time_interval
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-from homeassistant.util.async_ import run_callback_threadsafe
-from homeassistant.util.distance import convert
-import homeassistant.util.dt as dt_util
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback, EntityPlatform
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
 
 from .const import (
-    CONF_CIRCLES,
+    ATTR_ADDRESS,
+    ATTR_AT_LOC_SINCE,
+    ATTR_DRIVING,
+    ATTR_LAST_SEEN,
+    ATTR_PLACE,
+    ATTR_SPEED,
+    ATTR_WIFI_ON,
+    ATTRIBUTION,
     CONF_DRIVING_SPEED,
-    CONF_ERROR_THRESHOLD,
     CONF_MAX_GPS_ACCURACY,
-    CONF_MAX_UPDATE_WAIT,
-    CONF_MEMBERS,
-    CONF_SHOW_AS_STATE,
-    CONF_WARNING_THRESHOLD,
     DOMAIN,
+    LOGGER,
     SHOW_DRIVING,
-    SHOW_MOVING,
 )
 
-_LOGGER = logging.getLogger(__name__)
+_EXTRA_ATTRIBUTES = (
+    ATTR_ADDRESS,
+    ATTR_AT_LOC_SINCE,
+    ATTR_BATTERY_CHARGING,
+    ATTR_LAST_SEEN,
+    ATTR_PLACE,
+    ATTR_SPEED,
+    ATTR_WIFI_ON,
+)
 
-SPEED_FACTOR_MPH = 2.25
-EVENT_DELAY = timedelta(seconds=30)
 
-ATTR_ADDRESS = "address"
-ATTR_AT_LOC_SINCE = "at_loc_since"
-ATTR_DRIVING = "driving"
-ATTR_LAST_SEEN = "last_seen"
-ATTR_MOVING = "moving"
-ATTR_PLACE = "place"
-ATTR_RAW_SPEED = "raw_speed"
-ATTR_SPEED = "speed"
-ATTR_WAIT = "wait"
-ATTR_WIFI_ON = "wifi_on"
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the device tracker platform."""
+    account = hass.data[DOMAIN]["accounts"][entry.unique_id]
+    coordinator = account["coordinator"]
+    tracked_members = hass.data[DOMAIN]["tracked_members"]
+    logged_circles = hass.data[DOMAIN]["logged_circles"]
+    logged_places = hass.data[DOMAIN]["logged_places"]
+
+    @callback
+    def process_data(new_members_only: bool = True) -> None:
+        """Process new Life360 data."""
+        for circle_id, circle in coordinator.data["circles"].items():
+            if circle_id not in logged_circles:
+                logged_circles.append(circle_id)
+                LOGGER.info("Circle: %s", circle["name"])
```

Comment:
We should almost never log at info level in integrations. Please use debug level.

Reply:
Ok.